modules/acm/src/test/scala/edu/gemini/epics/acm/ObserveStateSpec.scala

```diff
@@ -0,0 +1,402 @@
+    (context.createChannel(_: String)).expects("gm:applyC.OMSS").returns(null)
+    val epicsService = new EpicsService(context)
+    val observe = new CaObserveSenderImpl(
+      "gmos::observeCmd",
+      "gm:apply",
+      "gm:applyC",
+      "gm:dc:observeC",
+      "gm:stop",
+      "gm:abort",
+      "GMOS Observe",
+      classOf[CarState],
+      epicsService)
+    // Start idle
+    assert(observe.applyState().isIdle())
+    // Post an observe
+    observe.post()
+
+    // CONFIGURATION
+    // APPLY Goes to busy
+    // VAL change
+    observe.onApplyValChange(4169)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4169)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+
+    // APPLY Goes to IDLE
+    // Another VAL change
+    observe.onApplyValChange(4169)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4169)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.IDLE)
+    assert(!observe.applyState().isIdle())
+
+    // OBSERVE
+    // OBSERVE goes BUSY
+    // VAL change
+    observe.onApplyValChange(4170)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4170)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+    // Another VAL change
+    observe.onApplyValChange(4170)
+    assert(!observe.applyState().isIdle())
+    // Observe CAR VAL change
+    observe.onObserveCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4170)
+    assert(!observe.applyState().isIdle())
+    // Apply goes IDLE
+    observe.onCarValChange(CarState.IDLE)
+    assert(!observe.applyState().isIdle())
+
+    // PAUSE observations
+    observe.onApplyValChange(4171)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4171)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+    // Another VAL change
+    observe.onApplyValChange(4171)
+    assert(!observe.applyState().isIdle())
+    // Observe CAR VAL change
+    observe.onObserveCarValChange(CarState.PAUSED)
+    // We are now idle
+    assert(observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4171)
+    assert(observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.IDLE)
+    assert(observe.applyState().isIdle())
+
+    // RESUME OBSERVE
+    observe.onApplyValChange(4172)
+    // TODO Fix why this is idle here, it should be busy
+    assert(observe.applyState().isIdle())
```

Comment:
@jluhrs I think this is wrong, we should go back to not idle when resumed, right?

Reply:
Observe and resume work almost the same. Resume will not succeed if the observation was not paused. The same post is used to trigger the resume, and from then it should follow the same sequence of events.

Reply:
Perhaps I need to send another post?